clients/fides-js/src/lib/providers/event-context.tsx

```diff
@@ -0,0 +1,94 @@
+import { createContext, h } from "preact";
+import { MutableRefObject, ReactNode } from "preact/compat";
+import { useCallback, useContext, useMemo, useRef } from "preact/hooks";
+
+import { FidesCookie, ServingComponent } from "../consent-types";
+import {
+  dispatchFidesEvent,
+  FidesEventDetailsTrigger,
+  FidesEventExtraDetails,
+  FidesEventType,
+} from "../events";
+
+interface UseEventProps {
+  triggerRef: MutableRefObject<FidesEventDetailsTrigger>;
+  setTrigger: (newValue: FidesEventDetailsTrigger) => void;
+  servingComponentRef: MutableRefObject<ServingComponent | undefined>;
+  setServingComponent: (newValue: ServingComponent | undefined) => void;
+  dispatchFidesEventAndClearTrigger: (
+    type: FidesEventType,
+    fidesCookie: FidesCookie | undefined,
+    extraDetails?: FidesEventExtraDetails,
+  ) => void;
+}
+
+const EventContext = createContext<UseEventProps | Record<any, never>>({});
+
+export const EventProvider = ({ children }: { children: ReactNode }) => {
+  const triggerRef = useRef<FidesEventDetailsTrigger>();
+  const servingComponentRef = useRef<ServingComponent | undefined>();
+
+  const updateContextValue = useCallback(
+    (newValue: FidesEventDetailsTrigger) => {
+      triggerRef.current = newValue;
+    },
+    [],
+  );
+
+  const updateServingComponent = useCallback(
+    (newValue: ServingComponent | undefined) => {
+      servingComponentRef.current = newValue;
+    },
+    [],
+  );
+
+  const dispatchFidesEventAndClearTrigger = useCallback(
```

Comment:
This is just a convenience wrapper around `dispatchFidesEvent` which also clears the trigger when it's done, to keep the context clean.